packages/components/src/components/hds/link/types.ts

```diff
@@ -0,0 +1,37 @@
+import type { HdsInteractiveSignature } from '../interactive/types.ts';
+
+export enum HdsLinkIconPositionValues {
+  Leading = 'leading',
+  Trailing = 'trailing',
+}
+
+export type HdsLinkIconPositions = `${HdsLinkIconPositionValues}`;
+
+export enum HdsLinkColorValues {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+export type HdsLinkColors = `${HdsLinkColorValues}`;
+
+export enum HdsLinkStandaloneSizeValues {
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export type HdsLinkStandaloneSizes = `${HdsLinkStandaloneSizeValues}`;
+
+export interface HdsLinkStandaloneSignature {
+  Args: HdsInteractiveSignature['Args'] & {
+    icon: string;
+    text: string;
+    color?: HdsLinkColors;
+    href?: string;
+    iconPosition?: HdsLinkIconPositions;
+    isHrefExternal?: boolean;
+    isRouteExternal?: boolean;
+    size?: HdsLinkStandaloneSizes;
+  };
+  Element: HdsInteractiveSignature['Element'];
```

Comment:
I set this to the `HdsInteractiveSignature['Element']` as it uses the interactive component under the hood. I don't think this will ever be a button though. It does correctly return a type error if you don't use this though.